src/main/kotlin/racingcar/car/Car.kt

```diff
@@ -1,11 +1,25 @@
 package racingcar.car
 
 class Car(
-    var position: Int = START_POSITION,
-    private val canMove: () -> Boolean = {
-        (MINIMUM_POWER..MAXIMUM_POWER).random() >= ENGINE_THRESHOLD
-    },
+    position: Int = START_POSITION,
+    private val carName: Name,
+    private val canMove: () -> Boolean = defaultCanMove,
 ) {
+    constructor(name: String) : this(
+        carName = Name(name),
+    )
+
+    constructor(name: String, canMove: () -> Boolean) : this(
+        carName = Name(name),
+        canMove = canMove,
+    )
+
+    val name: String
+        get() = carName.value
+
+    var position: Int = position
+        private set
```

Comment:
I wanted to make it so that the position value can be changed inside the class but not outside, so I modified the code like this. I'm a bit unsure if this is the right way to write it in Kotlin, so I’m leaving a comment. The desired functionality is achieved, but the code has become longer, and the same field is repeated, so it might be a bit confusing (like whether to carefully check if there’s a `var` or not, etc.). Is this how it's commonly written? (Or is it because I started with Java? 🤔)